htsget-search/src/htsget/bam.rs

```diff
@@ -61,41 +62,61 @@ where
   }
 
   pub fn search(&self, query: Query) -> Result<Response> {
-    // TODO check class, for now we assume it is None or "body"
-
     let (bam_key, bai_key) = self.get_keys_from_id(query.id.as_str());
 
-    let bai_path = self.storage.get(&bai_key, GetOptions::default())?;
-    let bai_index = bai::read(bai_path).map_err(|_| HtsGetError::io_error("Reading BAI"))?;
-
-    let byte_ranges = match query.reference_name.as_ref() {
-      None => self.get_byte_ranges_for_all_reads(bam_key.as_str(), &bai_index)?,
-      Some(reference_name) if reference_name.as_str() == "*" => {
-        self.get_byte_ranges_for_unmapped_reads(bam_key.as_str(), &bai_index)?
+    match query.class {
+      None | Some(Class::Body) => {
+        let bai_path = self.storage.get(&bai_key, GetOptions::default())?;
+        let bai_index = bai::read(bai_path).map_err(|_| HtsGetError::io_error("Reading BAI"))?;
+
+        let byte_ranges = match query.reference_name.as_ref() {
+          None => self.get_byte_ranges_for_all_reads(bam_key.as_str(), &bai_index)?,
+          Some(reference_name) if reference_name.as_str() == "*" => {
+            self.get_byte_ranges_for_unmapped_reads(bam_key.as_str(), &bai_index)?
+          }
+          Some(reference_name) => self.get_byte_ranges_for_reference_name(
+            bam_key.as_str(),
+            reference_name,
+            &bai_index,
+            &query,
+          )?,
+        };
+        self.format_urls(query, &bam_key, byte_ranges, None)
       }
-      Some(reference_name) => self.get_byte_ranges_for_reference_name(
-        bam_key.as_str(),
-        reference_name,
-        &bai_index,
-        &query,
-      )?,
-    };
+      Some(Class::Header) => {
+        let byte_ranges = self.get_byte_ranges_for_header()?;
+        self.format_urls(query, &bam_key, byte_ranges, Some(Class::Header))
+      }
+    }
+  }
 
+  /// Format the urls from the query.
+  fn format_urls(
+    &self,
+    query: Query,
```

Comment:
Either you don't pass the class and get it from the query, or don't pass the query and add the format, but please don't mix them.

Option 1:
```rust
 fn format_urls(
    &self,
    query: Query,
    bam_key: &String,
    byte_ranges: Vec<BytesRange>,
  ) -> Result<Response> {
```

Option 2:
```rust
 fn format_urls(
    &self,
    format: Option<Format>,
    class: Option<Class>,
    bam_key: &String,
    byte_ranges: Vec<BytesRange>,
  ) -> Result<Response> {
```